### ISSUE TITLE
MultiWayIf parse error

### ISSUE THREAD

**Issue by josephcsible** [ndmitchell/hlint#850]
This code compiles fine with GHC, but hlint reports a parse error:
```haskell
{-# LANGUAGE MultiWayIf #-}

main = if | 1 < 2
          -> putStrLn "good"
          | otherwise -> putStrLn "bad"
```

This seems to be caused by a combination of both MultiWayIf and the weird layout, because weird layout with regular `case` works fine:
```haskell
main = case () of _ | 1 < 2
                    -> putStrLn "good"
                    | otherwise -> putStrLn "bad"
```

And so does `MultiWayIf` with regular layout:
```haskell
{-# LANGUAGE MultiWayIf #-}

main = if | 1 < 2 -> putStrLn "good"
          | otherwise -> putStrLn "bad"
```

**Comment by ndmitchell** [comment]
I'm going to blame this on haskell-src-exts, without doing the due diligence to actually check. We're nearly at the point of removing haskell-src-exts and switching exclusively to GHC, at which point I hope this goes away, but it not, we can revisit it then.

**Comment by ndmitchell** [comment]
Confirmed works with HEAD that doesn't use HSE for parsing, so fixed in the next release.
